src/main/java/com/example/bookstore/service/book/BookServiceImpl.java

```diff
@@ -60,9 +59,9 @@ public void deleteById(Long id) {
     @Override
     public List<BookDto> search(BookSearchParametersDto searchParametersDto) {
         Specification<Book> bookSpecification = bookSpecificationBuilder.build(searchParametersDto);
-        return bookRepository.findAll((Sort) bookSpecification)
+        return bookRepository.findAll(bookSpecification)
                 .stream()
-                .map(bookMapper::toDto)
+                .map((Object book) -> bookMapper.toDto((Book) book))
```

Comment:
Why do you need this changes?

Reply:
I found out that search endpoint doesn't work with previous code, it broke at the moment of casting to (Sort)